### DISCUSSION TITLE
Growing `.next/server/pages` directory on production with fallback: false

### DISCUSSION THREAD

**Discussion by bits-by-brandon** [vercel/next.js#19254] [Help]
# Bug report

## Describe the bug
Urls that result in 404's are creating new files in `.next/server/pages`, when it should be serving the single cached `404` page. We are running with `fallback: false`, and generating all pages at build-time.

I've noticed that removing the call to `getStaticProps` in our `404.js` solves this issue, but our use case requires us to read data from the filesystem to generate the 404 page.

### How to reproduce
0. Pull down https://github.com/brandondc741/next-issue-19254
0. Build using `next build`
0. Run using `next start`
0. Visit a 404 url such as `hello` or `test`
0. Observe `.next/server/pages` create html pages for these 404 urls. 

### Context
We noticed a huge spike in storage used in our next.js production instance. When looking into it, we saw that the `.next/server/pages` directory was filled with bot traffic urls that should of received 404 responses. This quickly consumed our entire 64GB storage capacity for our production server. Now we have thousands of bot produced urls in our `.next/server/pages` directory.

## Screenshots
<img width="232" alt="Screen Shot 2020-11-17 at 5 12 03 PM" src="https://user-images.githubusercontent.com/9118169/99469638-0a048180-28f8-11eb-9c71-98d57edb24be.png">

## System information

- OS: macOS
- Version of Next.js: 9.5.0
- Version of Node.js: 12.18.4
- Deployment: next build && next start

**Comment by timneutkens** [comment]
You'll want to add `notFound: true` for pages that are not found: https://nextjs.org/docs/basic-features/data-fetching#getstaticprops-static-generation

**Reply by bits-by-brandon** [reply]
Hi @timneutkens I had a mistake with the title of the issue. This situation happens with `fallback: false`. My understanding is that all routes not specified in `getStaticPaths` should all considered `notFound`. See the example provided at https://github.com/brandondc741/next-issue-19254

From the docs:

>Note: notFound is not needed for fallback: false mode as only paths returned from getStaticPaths will be pre-rendered.
